jsapp/js/components/common/koboDropdown.es6

```diff
@@ -0,0 +1,202 @@
+    window.removeEventListener('click', this.onWindowClick);
+    this.cancelEscKeyListener();
+    this.cancelOutsideClickListener();
+  }
+
+  registerEscKeyListener() {
+    document.addEventListener('keydown', this.onAnyKeyWhileOpen);
+  }
+
+  cancelEscKeyListener() {
+    document.removeEventListener('keydown', this.onAnyKeyWhileOpen);
+  }
+
+  onAnyKeyWhileOpen(evt) {
+    if (
+      evt.key === 'Escape' ||
+      evt.code === 'Escape' ||
+      evt.keyCode === KEY_CODES.ESC ||
+      evt.which === KEY_CODES.ESC
+    ) {
+      this.hideMenu();
+    }
+  }
+
+  registerOutsideClickListener() {
+    window.addEventListener('click', this.checkOutsideClick);
+    window.addEventListener('touch', this.checkOutsideClick);
+    window.addEventListener('touchstart', this.checkOutsideClick);
+    window.addEventListener('contextmenu', this.checkOutsideClick);
+  }
+
+  cancelOutsideClickListener() {
+    window.removeEventListener('click', this.checkOutsideClick);
+    window.removeEventListener('touch', this.checkOutsideClick);
+    window.removeEventListener('touchstart', this.checkOutsideClick);
+    window.removeEventListener('contextmenu', this.checkOutsideClick);
+  }
+
+  checkOutsideClick(evt) {
+    let isOutsideClick = true;
+    const dropdownWrapperEl = ReactDOM.findDOMNode(this);
+
+    let loopEl = evt.target;
+    // If we check too much parents going upward looking for dropdown then we
+    // assume it is not a dropdown.
+    let parentsLookupLimit = 6;
+    while (loopEl.parentNode !== null) {
+      if (--parentsLookupLimit === 0) {break;}
+      loopEl = loopEl.parentNode;
+      if (
+        loopEl &&
+        typeof loopEl.isEqualNode === 'function' &&
+        loopEl.isEqualNode(dropdownWrapperEl)
+      ) {
+        isOutsideClick = false;
+        break;
+      }
+    }
+
+    if (isOutsideClick) {
+      this.hideMenu();
+    }
+  }
+
+  getWrapperModifiers() {
+    const wrapperMods = [];
+
+    if (
+      this.props.theme &&
+      typeof KOBO_DROPDOWN_THEMES[this.props.theme] !== 'undefined'
+    ) {
+      wrapperMods.push(this.props.theme);
+    } else {
+      wrapperMods.push(KOBO_DROPDOWN_THEMES.light);
+    }
+
+    if (
+      this.props.placement &&
+      typeof KOBO_DROPDOWN_PLACEMENTS[this.props.placement] !== 'undefined'
+    ) {
+      wrapperMods.push(this.props.placement);
+    } else {
+      wrapperMods.push(KOBO_DROPDOWN_PLACEMENTS['down-center']);
+    }
+
+    // These modifiers are for styling purposes only, i.e. they don't have
+    // anything to do with menu being visible.
+    if (this.state.isMenuVisible) {
+      wrapperMods.push('menu-visible');
+    } else {
+      wrapperMods.push('menu-hidden');
+    }
+
+    if (this.props.isDisabled) {
+      wrapperMods.push('disabled');
+    }
+
+    return wrapperMods;
+  }
+
+  render() {
+    return (
+      <bem.KoboDropdown m={this.getWrapperModifiers()}>
+        <bem.KoboDropdown__trigger onClick={this.onTriggerClick}>
```

Comment:
I took a sneak peak at your usability improvements branch :P I noticed that the caret:
![image](https://user-images.githubusercontent.com/16762675/123185162-45a19400-d463-11eb-9244-c18ddb89994a.png)
is a little hard to click on (it looks like you have to click slightly above it) Maybe we should add a wrapper class before the `props` content with a minimum height and width?

Reply:
I thought about it and I decided it would be better for the `KoboDropdown` user to be responsible for the trigger clickability area. Otherwise the wrapper with minimum height/width might get in a way. I tried to have the component as generic as possible - that's why the `bem.KoboDropdown__menuButton` is an optional useful BEM not a requirement :)

Reply:
I might add some tweaks to `KoboDropdown` thought while I work on it on that branch and discover something :)